File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/MotorModule.java
package com.example.tutorial4_1dependentcomponent.di;

import com.example.tutorial4_1dependentcomponent.model.Motor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MotorModule {
    @Provides
    Motor provideMotor() {
        return  new Motor();
    }
}

File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/MotorModule.kt
package com.example.tutorial4_1dependentcomponent.di

import com.example.tutorial4_1dependentcomponent.model.Motor
import dagger.Module
import dagger.Provides

@Module
class MotorModule {

    @Provides
    fun provideMotor(): Motor {
        return Motor()
    }
}